lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -150,7 +150,9 @@ async function resolveRestApiId() {
     }
     const apiName = apiGatewayResource
       ? apiGatewayResource.Properties.Name
-      : provider.apiName || `${this.options.stage}-${this.state.service.service}`;
+      : provider.apiName || (provider.apiGateway && provider.apiGateway.shouldStartNameWithService)
+        ? `${this.state.service.service}-${this.options.stage}`
```

Comment:
Instead of duplicating the logic, we should rather reuse the `naming.getApiGatewayName` utility - the reason for this bug is that it was not used previously I believe.